frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -142,141 +233,204 @@ function ProblemDisplay(props: ProblemDisplayParams) {
   };
 
   return (
-    <Content>
-      <MediumText>Name:</MediumText>
-      <TextInput
-        name="name"
-        value={newProblem.name}
-        onChange={handleChange}
-      />
-
-      <MediumText>Description:</MediumText>
-      <StyledMarkdownEditor
-        defaultValue={newProblem.description}
-        onChange={(getNewValue) => handleDescriptionChange(getNewValue())}
-      />
-
-      <MediumText>Difficulty:</MediumText>
-      {Object.keys(Difficulty).map((key) => {
-        const difficulty = Difficulty[key as keyof typeof Difficulty];
-        if (difficulty !== Difficulty.Random) {
-          return (
-            <DifficultyButton
-              onClick={() => handleEnumChange('difficulty', difficulty)}
-              active={difficulty === newProblem.difficulty}
-              enabled
-            >
-              {key}
-            </DifficultyButton>
-          );
-        }
-        return null;
-      })}
-
-      <MediumText>Problem Inputs:</MediumText>
-      {newProblem.problemInputs.map((input, index) => (
-        <div>
-          <Text bold>{`Input ${index + 1}`}</Text>
-          <TextInput
-            value={newProblem.problemInputs[index].name}
-            onChange={(e) => handleInputChange(index,
-              e.target.value, newProblem.problemInputs[index].type)}
+    <>
+      <MainContent>
+        <FlexBareContainer>
+          <SmallHeaderText>Problem</SmallHeaderText>
+          <InvertedSmallButtonLinkAutoLeftMargin
+            onClick={() => onClick(newProblem)}
+            to="/problems/all"
+          >
+            Back
+          </InvertedSmallButtonLinkAutoLeftMargin>
+          <SmallButton
+            onClick={() => onClick(newProblem)}
+          >
+            {actionText}
+          </SmallButton>
+        </FlexBareContainer>
+        <SettingsContainerHighPadding>
+          <PureTextInputTitle
+            placeholder="Write a nice title"
+            name="name"
+            value={newProblem.name}
+            onChange={handleChange}
+          />
+          <TitleDescriptionSeparator />
+          <StyledMarkdownEditor
+            placeholder="Write a nice description"
+            defaultValue={newProblem.description}
+            onChange={(getNewValue) => handleDescriptionChange(getNewValue())}
           />
-          <br />
-
-          {Object.keys(ProblemIOType).map((key) => {
-            const inputType = ProblemIOType[key as keyof typeof ProblemIOType];
-            return (
-              <ProblemIOTypeButton
-                onClick={() => handleInputChange(index,
-                  newProblem.problemInputs[index].name, inputType)}
-                active={inputType === newProblem.problemInputs[index].type}
+        </SettingsContainerHighPadding>
+
+        {editMode
+          ? (
+            <>
```

Comment:
Out of curiosity, what does this syntax do?

Reply:
The first is a ternary operator which I'm sure you know, the `<>` is a [React Fragment](https://reactjs.org/docs/fragments.html#short-syntax) so you can return a list of children without adding extra nodes to the DOM.